You-Can-Code-It/fndr
# Seeding data

# What this pull request does

## What

- All companies shall receive a CUID, NOT automatically, but hardcoding it
- The data is seeded to the database, so we get the necessary input to start moving forward with the application

## Why

- Seeding the data with a hardcoded CUID will ensure that different environments use the same id (not randomly created every time the data is seeded), so other features relying on a company's id will not fail due to different ids
- Having the companies' data in the database allows the application to go further since these resources can be used in the application

## How

- A file called "addIdsToCompanies.ts" is created to map over the .json file comprising all the companies' data, and to add to each object a CUID (unique ID). The library "cuid" is used to generate the ids
- It uses "fs.writeFileSync" function, so when the script is run, a new .json file is created in the specified folder
- A file called "seed.ts" is added in the "prisma" folder. It uses "PrismaClient" and the "createMany" method to insert the data into the render database. 
- The "company" object is destructured to contemplate eventual differences between the .json file and the model definition (e.g. "website" in the model versus "websiteUrl" in the .json). It also overwrites the id as mentioned
- This script is run with the `npx ts-node` command

## Steppings to test this

- Pull this branch in your machine
- Checkout to it
- Create a `.env` file in the root of your project following the `.env.example` file to accordingly setting your database up
- Run the `npx run seed` script in your terminal to seed the data!
- Check it out e.g. by opening the prisma interface (`npx prisma studio`)

## Anything else

- databases require special attention to set them up in different environments. So keep it in mind when testing this feature
